test/helpers/test_provisioner.rb

```diff
@@ -68,6 +68,17 @@ def deploy_metric_server
         found = kubectl.run("get", "-f", resource, log_failure: false).last.success?
         kubectl.run("create", "-f", resource) unless found
       end
+
+      # The service is often slow to start leading to randomly failed HPA tests
+      3.times do |round|
+        service = kubectl.run("-n", "kube-system", "get", "APIService", "v1beta1.metrics.k8s.io", "-o", "json")
```

Comment:
Minor nit but I would prefer if the L-values here would be split into separate variables instead of just one. I had to dig through the `kubectl.run` code to see what `service.first/last` were.